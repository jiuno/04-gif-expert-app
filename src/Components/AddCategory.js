import React, { useState } from 'react';
import PropTypes from 'prop-types';


export  const AddCategory = ({SetCategories}) => {
    
        const  [InputValue, setInputValue] = useState('');

        const handleInputChange = (e) => {
            
            setInputValue(e.target.value)
        }
        
        const handleSubmit = (e) => {
            e.preventDefault();
            if (InputValue.trim().length >= 2) {
                SetCategories((cats) => [InputValue, ...cats]); //Esto es categories ahora
                setInputValue(''); 
                
            }


            // console.log('Submit hecho');

        }
        return (
            <form onSubmit= {handleSubmit}>
                <input
                type="text" 
                value={InputValue}
                onChange={handleInputChange}
                
                />
            </form>
        )
    }

AddCategory.propTypes = {
    SetCategories: PropTypes.func.isRequired
}

