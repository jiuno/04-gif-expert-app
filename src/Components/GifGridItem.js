import React from 'react'

export const GifGridItem = ({id, title, url}) => {
    
    // console.log(url);
    
    return (
        <div className="card animate__animated animate__bounceIn ">
            <img src={url} alt="GIF" ></img>
            <hr></hr>
            <h3> {title} </h3>
        </div>
    )
}
