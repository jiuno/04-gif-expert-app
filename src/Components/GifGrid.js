import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs( category );

    return (
        <>
        <h3> {category} </h3>

        <div className="Card-Grid">
           
                {loading && <p className="animate__animated animate__heartBeat">Loading...</p> }
                { images.map( (img) =>  (
                <GifGridItem 
                { ...img} //Manda props.img con todas sus propiedades, no solo id, title y url.
                key={img.id}
                />))}

        </div>
        </>
    )
}
