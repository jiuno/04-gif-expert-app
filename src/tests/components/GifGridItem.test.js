// import React from 'react';
import {shallow} from 'enzyme';
import {GifGridItem} from '../../Components/GifGridItem'

describe('Pruebas para GifGridItem', () => {

    const title= 'Un titulo';

    const url='https://www.paginaweb.com';

    const wrapper = shallow(<GifGridItem title={title} url={url} /> );

    
    test('debe renderizar el componente', () => {

        expect( wrapper).toMatchSnapshot();
        
    });
    test('debe mostrar un parrafo con `title`.', () => {
        const h3 = wrapper.find('h3');      
          
        expect(h3.text().trim()).toBe( title );
    
    });

    test('Debe tener la img src y el alt al `url` suministrado en los props.', () => {
        
        const img = wrapper.find('img');

        expect( img.props().src ).toBe(url);
        expect( img.props().alt).toBe(title);
    })

    test('El div debe tener la clase `card animate__animated animate__bounceIn`', () => {
        
        const div = wrapper.find('div');

        expect(div.hasClass('card animate__animated animate__bounceIn')).toBe(true);
    })
    
    
        

    
})
