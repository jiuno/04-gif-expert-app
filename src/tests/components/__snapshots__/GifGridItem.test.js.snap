// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifGridItem debe renderizar el componente 1`] = `
<div
  className="card animate__animated animate__bounceIn "
>
  <img
    alt="Un titulo"
    src="https://www.paginaweb.com"
  />
  <hr />
  <h3>
     
    Un titulo
     
  </h3>
</div>
`;
